discovery/discovery.go

```diff
@@ -17,7 +17,9 @@ import (
 	"github.com/golang/glog"
 )
 
-var getOrchestratorsTimeoutLoop = 3 * time.Second
+const MinWorkingSetSize = 8
```

Comment:
Looks like we're introducing a min working set size here to determine when to pop Os off the suspension queue. I think an alternative that would avoid explicitly introducing a min working set size variable/function is the following:

---

Make the following update to the discovery loop:

```golang
// This loop runs until either all orchestrators have responded or a timeout
for i := 0; i < numAvailableOrchs && !timeout; i++ {
    // ...
}
```

Then, only the caller of `GetOrchestrators()` needs to be aware of the min working set size (which is the case right now outside of this PR) instead of making the `GetOrchestrators()` function aware of a min working set size variable/function.

The end result should be the same as what is currently implemented in this PR. The caller specifies the min working set size via `numOrchestrators` and `GetOrchestrators()` will fetch as many responses as possible from all Os, but if a bunch of Os are suspended such that the number of non-suspended Os is less than `numOrchestrators`, we'll pop Os off the suspension queue. The caller would be BroadcastSessionsManager which sets its `numOrchs` field to the min working set size which is passes as `numOrchestrators` in the `GetOrchestrators()` call.

WDYT?

Reply:
What you mention makes sense, the main reason I went with this (and I'm open to alternatives) is because if we select as many O's as possible e.g. 50 , and 5 don't respond. We'll still have a working set of 45 which seems sufficient enough to have to pop on suspended O's until there's 50 O's again. It might make more sense to do that only when the working set would otherwise get too small.

Reply:
Right. So if `numOrchestrators` reflected a min working set size that is smaller than 50, which is the # of available Os in this scenario, defined in BroadcastSessionsManager then we wouldn't pop any Os from the suspension queue in the scenario that you described.